pkg/controllers/termination/terminate.go

```diff
@@ -61,6 +61,14 @@ func (t *Terminator) cordon(ctx context.Context, node *v1.Node) error {
 	// 2. Cordon node
 	persisted := node.DeepCopy()
 	node.Spec.Unschedulable = true
+	// Handle nil map
+	if node.Labels == nil {
```

Comment:
Personal preference: you can take it or leave it. I prefer,

```go
if node.Labels == nil {
  node.Labels = map[string]string{}
}
node.Labels["node.kubernetes.io/exclude-from-external-load-balancers"] = "karpenter"
```

so that the map indexing isn't repeated

Reply:
Good call. Updated

Reply:
optional syntactic sugar to do this.

`node.Labels = lo.Assign(node.Labels, map[string]string{"key":"value"})`